addons/pinniped/config-controller/controllers/controllers.go

```diff
@@ -158,17 +146,41 @@ type concierge struct {
 	Audience string `yaml:"audience,omitempty"`
 }
 
-// nolint:funlen,nolintlint // Eh, we can live with a function of this length
-func (c *PinnipedController) reconcileAddonSecret(ctx context.Context, cluster *clusterapiv1beta1.Cluster, pinnipedInfoCM *corev1.ConfigMap) error {
-	log := c.Log.WithValues(constants.NamespaceLogKey, cluster.Namespace, constants.NameLogKey, cluster.Name)
-	// check if cluster is scheduled for deletion, if so, delete addon secret on mgmt cluster
-	if !cluster.GetDeletionTimestamp().IsZero() {
-		log.V(1).Info("cluster is getting deleted, deleting addon secret")
-		if err := c.reconcileClusterDelete(ctx, client.ObjectKeyFromObject(cluster)); err != nil {
-			return fmt.Errorf("failed to reconcile cluster delete: %w", err)
+// TODO: simplify this function
```

Comment:
+1 for simplify & smaller helpers but cool with a todo for later also!

Reply:
yeah, I think I might do the refactor now since I hate the TODO :/